lib/repl.js

```diff
@@ -1521,12 +1521,161 @@ function isRecoverableError(e, code) {
       return pos > 0 && frames[pos - 1].length === frames[pos].length;
     }
 
+    if (message.startsWith('Unexpected token'))
```

Comment:
In version v8.11.3, without this particular check, things works. I'm wondering, now with this code in place, we are sort of re-doing the check (may be?). May be I'm wrong here too. 

Actually as discussed in the issue, this change broken the partial JSON stuff. The diff is over here:
https://github.com/nodejs/node/commit/ebfa8b14645798b486e010e3f8bcec693f42cd6f#diff-293fefe850253c8c28e2481c0b7f76e2L647

As seen in the comments, looks like partial template literal also broken in latest version. For example:
```
`io.js ${a
... }`
```

works in v8/9. But broken in v10x.